### PR TITLE
Warnint util 4516 v5

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4516

Describe changes:
- Fix integer warnings `-Wimplicit-int-conversion` in all files

Extends #6792 : complete with CI check...

Replaces #6820 with CIFuzz fix for defrag

Thoughts about the commit history ?
This PR can be split into smaller PRs as some points need a careful review (detect-content.c for instance)

### CODE REVIEW

**Comment by victorjulien** [comment]
Converted to draft due to "fixup" commit.

File: src/util-decode-mime.c [hunk 1]
```diff
@@ -686,7 +686,11 @@ static uint8_t * GetFullValue(DataValue *dv, uint32_t *len)
 static inline uint8_t * FindBuffer(const uint8_t *src, uint32_t len, const uint8_t *find, uint32_t find_len)
 {
     /* Use utility search function */
-    return BasicSearchNocase(src, len, find, find_len);
+    if (find_len > UINT16_MAX) {
+        // TODO rewrite all this to have app-layer events
```
Comment (victorjulien) [inline src/util-decode-mime.c:690]: ticket

**Comment by catenacyber** [comment]
> Converted to draft due to "fixup" commit.

Ah I had indeed forgotten to fix it up in its real commit

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on build_asan.
  

Pipeline 6404

File: src/util-decode-mime.c (hunk 1 above)
Comment (catenacyber) [reply src/util-decode-mime.c:690]: Thanks for pointing this
I did https://github.com/OISF/suricata/pull/7074
I managed to get rid of this check at this level (like `find_len` is always small enough) so I do not think we need a ticket for this

**Comment by catenacyber** [comment]
Status : splitting this in multiple PRs
- #7073 easy fix (`tolower`)
- #7074 complex fix (util-decode-mime)

Waiting to rebase on what gets merged
Keeping this open with the rest of the code changes to get, and comments to take into account

File: src/flow.h [hunk 2]
```diff
@@ -580,7 +580,7 @@ int FlowGetPacketDirection(const Flow *, const Packet *);
 
 void FlowCleanupAppLayer(Flow *);
 
-void FlowUpdateState(Flow *f, enum FlowState s);
+void FlowUpdateState(Flow *f, const FlowStateType s);
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply src/flow.h:583]: What do you expect from the compiler ?

With current code (ie `enum FlowState s`)
`FlowUpdateState(p->flow, 123456);` gets no warning on my setup

(And I do not get more warnings with this change, like clang is ok to pass 123456 to an uint16_t )

[the remark this answers is not in the document]
Comment (catenacyber) [reply src/flow.h:583]: Tried something here https://github.com/OISF/suricata/pull/7120

**Comment by catenacyber** [comment]
Finally replaced by https://github.com/OISF/suricata/pull/7470

**Pull request closed without merging**